modules/math/src/test/java/com/opengamma/strata/math/impl/interpolation/NonnegativityPreservingCubicSplineInterpolatorTest.java

```diff
@@ -268,265 +275,281 @@ public void flipMultiTest() {
-    interpPos.interpolate(xValues, yValues);
+    assertThatIllegalArgumentException()
+        .isThrownBy(() -> interpPos.interpolate(xValues, yValues));
   }
 
   /**
    * 
    */
-  @Test(expectedExceptions = IllegalArgumentException.class)
+  @Test
   public void diffDataTest() {
     final double[] xValues = new double[] {1., 2., 3., 4. };
     final double[] yValues = new double[] {0., 0.1, 0.05 };
 
     PiecewisePolynomialInterpolator interp = new NaturalSplineInterpolator();
     PiecewisePolynomialInterpolator interpPos = new NonnegativityPreservingCubicSplineInterpolator(interp);
-    interpPos.interpolate(xValues, yValues);
+    assertThatIllegalArgumentException()
+        .isThrownBy(() -> interpPos.interpolate(xValues, yValues));
   }
 
   /**
    * 
    */
-  @Test(expectedExceptions = IllegalArgumentException.class)
+  @Test
   public void diffDataMultiTest() {
     final double[] xValues = new double[] {1., 2., 3., 4. };
     final double[][] yValues = new double[][] { {2., 0., 0.1, 0.05, 2. }, {1., 0., 0.1, 1.05, 2. } };
 
     PiecewisePolynomialInterpolator interp = new NaturalSplineInterpolator();
     PiecewisePolynomialInterpolator interpPos = new NonnegativityPreservingCubicSplineInterpolator(interp);
-    interpPos.interpolate(xValues, yValues);
+    assertThatIllegalArgumentException()
+        .isThrownBy(() -> interpPos.interpolate(xValues, yValues));
   }
 
   /**
    * 
    */
-  @Test(expectedExceptions = IllegalArgumentException.class)
+  @Test
   public void nullXdataTest() {
-    double[] xValues = new double[] {1., 2., 3., 4. };
+    double[] xValues = null;
     double[] yValues = new double[] {0., 0.1, 0.05, 0.2 };
-    xValues = null;
 
     PiecewisePolynomialInterpolator interp = new CubicSplineInterpolator();
     PiecewisePolynomialInterpolator interpPos = new NonnegativityPreservingCubicSplineInterpolator(interp);
-    interpPos.interpolate(xValues, yValues);
+    assertThatIllegalArgumentException()
+        .isThrownBy(() -> interpPos.interpolate(xValues, yValues));
   }
 
   /**
    * 
    */
-  @Test(expectedExceptions = IllegalArgumentException.class)
+  @Test
   public void nullYdataTest() {
     double[] xValues = new double[] {1., 2., 3., 4. };
-    double[] yValues = new double[] {0., 0.1, 0.05, 0.2 };
-    yValues = null;
+    double[] yValues = null;
 
     PiecewisePolynomialInterpolator interp = new CubicSplineInterpolator();
     PiecewisePolynomialInterpolator interpPos = new NonnegativityPreservingCubicSplineInterpolator(interp);
-    interpPos.interpolate(xValues, yValues);
+    assertThatIllegalArgumentException()
+        .isThrownBy(() -> interpPos.interpolate(xValues, yValues));
   }
 
   /**
    * 
    */
-  @Test(expectedExceptions = IllegalArgumentException.class)
+  @Test
   public void nullXdataMultiTest() {
-    double[] xValues = new double[] {1., 2., 3., 4. };
+    double[] xValues = null;
```

Comment:
This deliberate?

Reply:
Code moved to a lambda, but local variables can't be mutated, so need to nullify at source